app/services/saml_request_validator.rb

```diff
@@ -71,6 +73,17 @@ def ial2_context_requested?
     end
   end
 
+  def ial_max_requested?
+    case authn_context
+    when Array
+      authn_context.any? do |classref|
+        Saml::Idp::Constants::IALMAX_AUTHN_CONTEXT_CLASSREF == classref
+      end
+    else
+      Saml::Idp::Constants::IALMAX_AUTHN_CONTEXT_CLASSREF == classref
+    end
```

Comment:
ok I have a pitch here, if `authn_context` is either a string or an array of strings, we can do....
```suggestion
    Array(authn_context).include?(Saml::Idp::Constants::IALMAX_AUTHN_CONTEXT_CLASSREF)
```
because of this neat trick:
```ruby
Array('a')
=> ["a"]
Array(['a', 'b'])
=> ["a", "b"]
```

Reply:
very into it